linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -3072,7 +3099,46 @@ private[optimizer] abstract class OptimizerCore(config: CommonPhaseConfig) {
 
     (op: @switch) match {
       case === | !== =>
-        val positive = (op == ===)
+        // Try to optimize as a primitive JS strict equality
+
+        def canBePrimitiveNum(tpe: RefinedType): Boolean = tpe.base match {
+          case AnyType | ByteType | ShortType | IntType | FloatType | DoubleType =>
+            true
+          case ClassType(cls) =>
+            !tpe.isExact && MaybeHijackedPrimNumberClasses.contains(cls)
+          case _ =>
+            false
+        }
+
+        def isWhole(tpe: RefinedType): Boolean = tpe.base match {
+          case ByteType | ShortType | IntType =>
+            true
+          case ClassType(cls) =>
+            cls == Definitions.BoxedByteClass ||
+            cls == Definitions.BoxedShortClass ||
+            cls == Definitions.BoxedIntegerClass
+          case _ =>
+            false
+        }
```

Comment:
To check my understanding: We do not have longs here, because we do not have longs in `canBePrimitiveNum`, because we never represent them with primitive numbers. Correct?